Add sagas, server functions, tests fo reducers, sagas

Safwood opened the pull request:
![Screenshot_3](https://user-images.githubusercontent.com/69139691/102471813-b3a97080-4066-11eb-89d1-4f26bf719d75.jpg)
![Screenshot_4](https://user-images.githubusercontent.com/69139691/102471816-b4da9d80-4066-11eb-9de3-27610cce149c.jpg)
Реализованы возможности регистрации, отправки и получения данных банковской карты, получения списка адресов, маршрутов. Все запросы на сервер выполняются с помощью саг. На странице с картой добавлено окно с формой. Написаны тесты для саг и редьюсеров.  Если данные кредитной карты заполнены - сразу показывается форма выбора маршрута. При вводе адресов в выпадающих списках выполняется фильтрация.

Safwood committed:
sagas

Safwood committed:
sagas1

Safwood committed:
check mistakes

Safwood committed:
check mistakes1

Safwood committed:
check mistakes3

Safwood committed:
feat(components): add sagas, server function, tests

Replace middlewars for agas, add sagas for card, route, regstration. Add server requests, create
tests for sagas, reducers

Allarepossible commented on src/actions.js:
```diff
@@ -3,9 +3,23 @@ export const LOG_OUT = "LOG_OUT"
 export const AUTHENTICATE = "AUTHENTICATE"
```
можно разделить по разным файлам и вынести в папку /actions

Allarepossible commented on src/components/MapNotification.jsx:
```diff
@@ -0,0 +1,24 @@
+import React, {Component} from 'react';
+import { Link } from "react-router-dom"
+
+class MapNotification extends Component {
```
можно использовать функциональные компоненты - они проще для чтения

```
const MapNotification = () => (
   <div className="Map-page">
        <div className="Form-container Form-container--profile">
        <form className="Form">
          <div className="Form__notification-wrapper">
          <h2 className="Form__heading Form__heading--profile">Профиль</h2>
            <p className="Form__notification">
            Для заказа такси заполните платёжные данные в профиле. 
            </p>
            <Link to="/profile" className="Entry-button">Перейти в профиль</Link>
          </div>
          </form>
        </div>
      </div>
);
```

Allarepossible commented on src/components/OrderForm.jsx:
```diff
@@ -0,0 +1,115 @@
+        <form onSubmit={this.sendOrder} className="Form Form--order-form Order">
+          <div className="Order-addresses">
+            <div className="Order-addresses__address Order-addresses__address--from">
+              <select onChange={this.changeAddress1} className="Order-addresses__input "type="text" placeholder="Адрес отправления">
```
вот тут можно сделать  доп пустой option. и фильтрацию вынести из рендера
```
render() {
const addressList1 = this.props.addressList.length ? this.props.addressList.filter(item => item !== this.state.address2) : [];
    const addressList2 = this.props.addressList.length ? this.props.addressList.filter(item => item !== this.state.address1) : [];

    return (
      <div className="Form-container Form-container--order-form">
        <form onSubmit={this.sendOrder} className="Form Form--order-form Order">
          <div className="Order-addresses">
            <div className="Order-addresses__address Order-addresses__address--from">
              <select onChange={this.changeAddress1} className="Order-addresses__input" value={this.state.address1}>
                <option></option>
                { addressList1.map (item => <option key={item} value={item}>{item}</option>)}
              </select>
              <button className="Order-addresses__button">
                <img className="Order-addresses__button-image" src={cross} alt="cross"/>
              </button>
              <button className="Order-addresses__button Order-addresses__button--errow">
                <img  className="Order-addresses__button-image" src={errowDown} alt="errowDown"/>
              </button>
            </div>
            <div className="Order-addresses__address Order-addresses__address--to">
              <select onChange={this.changeAddress2} className="Order-addresses__input" value={this.state.address2}>
                <option></option>
              { addressList2.map (item => <option key={item} value={item}>{item}</option>)}
              </select>
              ...
```

Safwood committed:
feat(components): correct mistakes, add re-order taxi

Replace middlewars for agas, add sagas for card, route, regstration. Add server requests,
create
tests for sagas, reducers
